# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1421788758 -3600
#      Tue Jan 20 22:19:18 2015 +0100
# Branch gui-release
# Node ID b6bd9159c1c9c4b6e6fba800bbc39993d69442b2
# Parent  be7ac98fab43e2cae6d174437bb8840f5f014fa7
check existence of files when restoring last session (bug #44054)

* file-editor.cc (construct): check whether files from last session exist
  before call request to open it

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -1562,17 +1562,21 @@ file_editor::construct (void)
   //restore previous session
   if (settings->value ("editor/restoreSession", true).toBool ())
     {
       QStringList sessionFileNames
         = settings->value ("editor/savedSessionTabs",
                            QStringList ()).toStringList ();
 
       for (int n = 0; n < sessionFileNames.count (); ++n)
-        request_open_file (sessionFileNames.at (n));
+        {
+          QFileInfo file = QFileInfo (sessionFileNames.at (n));
+          if (file.exists ())
+            request_open_file (sessionFileNames.at (n));
+        }
     }
 
   check_actions ();
 }
 
 void
 file_editor::add_file_editor_tab (file_editor_tab *f, const QString& fn)
 {
